### PR TITLE
[Review] Generate pretty formatted config file and sort with localname

### LINKED ISSUE #57: pretty formatted yaml file
Using pyaml?
https://pypi.python.org/pypi/pyaml
```
> print pyaml.dump(data)
- git:
    local-name: vcstools/wstool
    uri: https://github.com/vcstools/wstool.git
- git:
    local-name: vcstools/vcstools
    uri: https://github.com/vcstools/vcstools.git
    version: master
- git:
    local-name: wkentaro/inbox
    uri: git@github.com:wkentaro/inbox.git
    version: master
- git:
    local-name: wkentaro/pycd
    uri: git@github.com:wkentaro/pycd.git
    version: master
- git:
    local-name: wkentaro/tips
    uri: git@github.com:wkentaro/tips.git
    version: master
...
```

Also I hope the keys (uri, version ..) should appear even if no value.

This makes it easy to read & edit .rosinstall.

### PR DESCRIPTION
**Pull request by wkentaro**
Closes #57

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def run(self):
       data_files=data_files,
       cmdclass=cmdclass,
       # rosinstall dependency to be kept in order not to break ros hydro install instructions
-      install_requires=['vcstools>=0.1.37', 'pyyaml'],
+      install_requires=['vcstools>=0.1.37', 'pyyaml', 'pyaml'],
```
Comment (tkruse) [inline setup.py]: I had not noticed before. Is there any meaningful difference between pyyaml and pyaml that is important to the feature? I looked at : https://pypi.python.org/pypi/pyaml/15.8.2
If possible, less dependencies are better.

File: src/wstool/config_yaml.py [hunk 2]
```diff
@@ -386,7 +385,8 @@ def generate_config_yaml(config, filename, header):
     with open(os.path.join(config.get_base_path(), filename), 'w+b') as f:
         if header is not None:
             f.write(header.encode('UTF-8'))
-        items = [x.get_legacy_yaml() for x in config.get_source()]
+        items = [x.get_legacy_yaml() for x in
+                 sorted(config.get_source(), key=lambda x:x.get_local_name())]
```
Comment (tkruse) [inline src/wstool/config_yaml.py]: I believe I said that rosws must not be affected. Sorting here probably affects rosws, no?

**Comment by tkruse** [comment]
Overall I would say this should not be merged before we know rosws is not affected after depending on the next version of wstool.

File: setup.py (hunk 1 above)
Comment (wkentaro) [inline setup.py]: pyaml is for pretty formatted yaml.

Comment (wkentaro) [inline setup.py]: pyaml is necessary to write yaml with pretty format.

**Comment by wkentaro** [comment]
I squashed the commit with test fixes.
Please review.

File: src/wstool/config_yaml.py (hunk 2 above)
Comment (wkentaro) [inline src/wstool/config_yaml.py]: it seems that rosinstall does not use wstool.config_yaml:generate_config_yaml.
```
ken at cobra in /…/local/lib/python2.7/site-packages/rosinstall on master…
% grep generate_config_yaml *
```

File: setup.py (hunk 1 above)
Comment (tkruse) [inline setup.py]: Please look at the text at https://pypi.python.org/pypi/pyaml/15.8.2, which describes pyaml is not strictly necessary, it just make pretty pyYaml even prettier.

Comment (wkentaro) [inline setup.py]: Thanks. I overlooked that.

**Comment by wkentaro** [comment]
The latest commit should not affect rosinstall, and does not depend on pyaml.
Thanks for the comment. @tkruse

**Comment by tkruse** [comment]
lgtm, though in the current state rosws and rosinstall won't pretty-format. Maybe that could be done for consistency (without sorting). But with the API, this can be done in the rosinstall repo independently.

**Comment by wkentaro** [comment]
so this pr status is waiting for confirming new wstool's stability?

> Overall I would say this should not be merged before we know rosws is not affected after depending on the next version of wstool.

**Comment by tkruse** [comment]
No, I think we can release when @wjwwood is ready. rosws tests might break with the pretty formatting, but that can be fixed later.

**Comment by wjwwood** [comment]
Code review looks good, so I'll go ahead and include this in the release I'm about to make.

**Pull request merged**